specification/keyvault/data-plane/Security.KeyVault.SecurityDomain/models.tsp

```diff
@@ -0,0 +1,226 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "@typespec/http";
+
+using Azure.Core;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+/**
+ * Operation status.
+ */
+@lroStatus
+union OperationStatus {
+  string,
+
+  /** The operation succeeded. */
+  @lroSucceeded
+  Success: "Success", // The correct response should be "Succeeded", but this has already shipped.
+
+  /** The operation is in progress. */
+  InProgress: "InProgress",
+
+  /** The operation failed. */
+  @lroFailed
+  Failed: "Failed",
+}
+
+/**
+ * The Security Domain operation status.
+ */
+model SecurityDomainOperationStatus {
+  /**
+   * Operation status.
+   */
+  status?: OperationStatus;
+
+  /**
+   * Details of the operation status.
+   */
+  @encodedName("application/json", "status_details")
+  statusDetails?: string;
+}
+
+/**
+ * The key vault error exception.
+ */
+@error
+model KeyVaultError {
+  /**
+   * The key vault server error.
+   */
+  @visibility("read")
+  error?: Error;
+}
+
+/**
+ * The key vault server error.
+ */
+model Error {
+  /**
+   * The error code.
+   */
+  @visibility("read")
+  code?: string;
+
+  /**
+   * The error message.
+   */
+  @visibility("read")
+  message?: string;
+
+  /**
+   * The key vault server error.
+   */
+  @visibility("read")
+  @encodedName("application/json", "innererror")
+  innerError?: Error;
+}
+
+/**
+ * The Security Domain download operation requires customer to provide N certificates (minimum 3 and maximum 10)
+ * containing a public key in JWK format.
+ */
+model CertificateInfoObject {
+  /**
+   * Certificates needed from customer.
+   */
+  @maxItems(10)
+  @minItems(3)
+  certificates: SecurityDomainJsonWebKey[]; // TODO: These need to be unique items
+
+  /**
+   * Customer to specify the number of certificates (minimum 2 and maximum 10) to restore Security Domain.
+   */
+  @maxValue(10)
+  @minValue(2)
+  required?: int32 = 2;
+}
+
+/**
+ * A JSON Web Key (JWK) for use in a security domain operation.
+ */
+model SecurityDomainJsonWebKey {
```

Comment:
Nit: seems we should be defining common types like JWKs in a common tsp further up. I won't block on this, but we should consider it. We don't want JWKs - an industry standard - diverging accidentally or incidentally.